src/openms/include/OpenMS/ANALYSIS/ID/FalseDiscoveryRate.h

```diff
@@ -114,6 +158,488 @@ namespace OpenMS
     /// calculates the FDR, given two vectors of scores
     void calculateFDRs_(std::map<double, double>& score_to_fdr, std::vector<double>& target_scores, std::vector<double>& decoy_scores, bool q_value, bool higher_score_better) const;
 
+    template<typename T> struct IsIDType
+    {
+      static bool const value = std::is_same<T, PeptideIdentification>::value || std::is_same<T, ProteinIdentification>::value;
+    };
+
+    template<typename T> struct IsHitType
+    {
+      static bool const value = std::is_same<T, PeptideHit>::value || std::is_same<T, ProteinHit>::value;
+    };
+    /*
+    void getPeptideScoresFromMap_(
+        std::vector<std::pair<double,bool>>& scores_labels,
+        const ConsensusMap & cmap,
+        bool all_hits,
+        int charge,
+        const String& identifier) const;
+
+    void getScores_(
+      std::vector<std::pair<double,bool>>& scores_labels,
+      const ProteinIdentification & id) const;
+
+    void getScores_(
+      std::vector<std::pair<double,bool>>& scores_labels,
+      const std::vector<PeptideIdentification> & ids,
+      bool all_hits,
+      int charge, const String& identifier) const;
+
+    void getScores_(
+        std::vector<std::pair<double,bool>>& scores_labels,
+        const PeptideIdentification& id,
+        bool all_hits,
+        int charge, const String& identifier) const;
+
+    void getScores_(
+      std::vector<std::pair<double,bool>>& scores_labels,
+      const std::vector<PeptideIdentification> & targets,
+      const std::vector<PeptideIdentification> & decoys,
+      bool all_hits,
+      int charge,
+      const String& identifier) const;
+      */
+
+    //TODO could be done with set of target accessions, too
+    //TODO even better: store nr targets and nr decoys when creating the groups!
+    //TODO alternative scoring is possible, too (e.g. ratio of tgts vs decoys),
+    // this requires templatization of the scores_labels vector though
+    void getScores_(
```

Comment:
maybe add some brief comment what and how PFDRs are calculated

Reply:
even better: do it in the class docu!

Reply:
done at the apply function and in class docu. IMHO does make less sense at the getScores_ function here as hroest suggested.